pkg/bundle/util.go

```diff
@@ -48,48 +48,52 @@ func bundleHasCondition(bundle *trustapi.Bundle, condition trustapi.BundleCondit
 // bundle object.
 // LastTransitionTime will not be updated if an existing condition of the same
 // Type and Status already exists.
-func (b *bundle) setBundleCondition(bundle *trustapi.Bundle, condition trustapi.BundleCondition) {
+func (b *bundle) setBundleCondition(conditions *[]trustapi.BundleCondition, generation int64, condition trustapi.BundleCondition) {
```

Comment:
I updated the PR based on what I found in some other projects.
What I did there is quite opposite from what you are asking unfortunately 😓.
I made all fields of the condition positional parameters because that way it was clear what fields are expected and what fields are not expected.

Reply:
What about two functions? One for creating the condition you want to be present as part of the reconcile logic ("constructor") and another for merging it into the bundle status?

Reply:
Thanks for the example. I updated the PR to match how https://github.com/kubernetes/apimachinery/blob/1629459297c77f725c47f7121851db2ee94572bd/pkg/api/meta/conditions.go#L30 works.